tests/cbmc/stubs/memset.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * FUNCTION: memset
+ *
+ * Override the version of memset used by CBMC.
+ * This takes advantage of the fact that 64bit operations require fewer array updates,
+ * which can make this version faster than the naive unrolling when used in CBMC.
+ * Benchmark your particular proof to know for sure.
+ */
+
+ #undef memset
+
+ #include <cbmc_proof/nondet.h>
+ #include <stddef.h>
+
+ /**
+  * Override the version of memset used by CBMC. Users may not want to pay
+  * for the cost of performing the computation of memset in proofs. In that
+  * case, this stub at least checks for the preconditions and make sure to
+  * havoc all elements pointed by *s up to n.
+  */
+ void *memset_impl(void *s, int c, size_t n) {
```

Comment:
the file is called "memset", but it doesn't do a proper memset. Name the file something that reflects what is actually going on.

Reply:
I renamed to `memset_havoc`.